pkg/reconciler/iploop.go

```diff
@@ -54,17 +55,22 @@ func NewReconcileLooperWithClient(ctx context.Context, k8sClient *kubernetes.Cli
 		return nil, logging.Errorf("failed to retrieve all IP pools: %v", err)
 	}
 
-	pods, err := k8sClient.ListPods(ctx)
-	if err != nil {
+	looper := &ReconcileLooper{
+		k8sClient:      *k8sClient,
+		requestTimeout: timeout,
+	}
+
+	// migrate the podRef format if needed
+	if err := looper.migrationPodRef(ctx, ipPools); err != nil {
```

Comment:
Ok, I think I get it how it triggers the migration, so here in the control loop, we check if the migration has happened and apply it here.